### PR TITLE
Added feature to show QR code for bug URLs

### LINKED ISSUE #636: [feature] Add QR code to bug URLs
For checking in mobile browsers it would be helpful to have a QR code (or button to show one) for the URLs of problematic sites.

### PR DESCRIPTION
**Pull request by tagawa**
Also added font icon so font files are updated.
Fixes #636

webcompat-qrcode

### CODE REVIEW

**Comment by magsout** [comment]
Could you prefix className with `wc-` plz

File: webcompat/static/css/development/components/issue-detail.css [hunk 1]
```diff
@@ -172,6 +172,35 @@
   content: "\f00d";
 }
 
+/* QR code launcher */
+.Qr-wrapper {
```
Comment (magsout) [inline webcompat/static/css/development/components/issue-detail.css]: Could you move all `Qrimage` style in `Qrimage.css` one className per file. 

One file = one Component

Comment (tagawa) [inline webcompat/static/css/development/components/issue-detail.css]: Sure, done. I copied the structure of the ```LabelEditor``` code so maybe that should be moved to a single file as well.

**Comment by tagawa** [comment]
Thanks for the quick feedback @magsout. I was keeping name consistency with ```LabelEditor```. Do you mean prefix all the classes with ```wc-```?

**Comment by magsout** [comment]
I started to add a prefix to all classes but I have not finished yet. Prefix all new component is a good thing

**Comment by magsout** [comment]
And yes prefix all class by `wc-ClassName`

**Comment by tagawa** [comment]
Thanks @magsout. Done.

File: webcompat/static/css/development/components/qrcode.css [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  display:inline-block;
+}
+@media (--viewport-375px) {
+  .wc-QrImage-wrapper {
+    position:static; /* position of label_editor not depend of this class */
+  }
+}
+
+/* QR code */
+.QrImage
```
Comment (magsout) [inline webcompat/static/css/development/components/qrcode.css]: Could you add prefix here too ?

**Comment by miketaylr** [comment]
Thanks Daniel. I'll try to pull this down locally and play with it either this afternoon or early tomorrow.

Ideally we want to write some tests for this--I'm happy to help with that.

File: webcompat/static/js/lib/qrcode.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  events: {
+    'click .wc-QrImage-launcher:not(.is-active)': 'showQr',
+    'click .wc-QrImage-launcher.is-active': 'closeQr'
+  },
+  keyboardEvents: {
+    'q': 'showQr'
+  },
+  render: function() {
+    // Extract page URL from bug description
+    var body = this.model.attributes.body;
```
Comment (miketaylr) [inline webcompat/static/js/lib/qrcode.js]: I think we probably want to use [`Model#get`](http://backbonejs.org/#Model-get) here. I don't think `attributes` is a public API. (I could be wrong though).

File: webcompat/static/js/lib/qrcode.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  template: _.template($('#qr-image-tmpl').html()),
+  render: function() {
+    var qrDataUrl = qr.toDataURL({value: this.url, size: 8});
+    this.$el.html(this.template({'qrDataUrl': qrDataUrl}));
+    return this;
+  },
+  closeQr: function() {
+    this.issueView.qrButton.removeClass('is-active');
+    // detach() (vs remove()) here because we don't want to lose events if the
+    // user reopens the editor.
```
Comment (miketaylr) [inline webcompat/static/js/lib/qrcode.js]: s/editor/qr code/

File: webcompat/templates/issue.html [hunk 5]
```diff
@@ -127,9 +132,17 @@ <h1 class="wc-IssueDetail-title wc-Title--l">
...
+  <script type="text/template" id="qr-image-tmpl">
+    <div class="wc-QrImage-header">
+      <button class="wc-QrImage-btn">Close</button>
+    </div>
+    <div class="wc-QrImage-main"><img src="<%= qrDataUrl %>" alt="QR code"></div>
+  </script>
+  
 {% endblock %}
 {% block extrascripts %}
+<script src="{{ url_for('static', filename='js/vendor/qr.min.js')|bust_cache }}"></script>
```
Comment (miketaylr) [inline webcompat/templates/issue.html]: Can we go ahead and add the qr.min.js file to the Grunt concat task?

We talked about lazy-loading this script in the future, but for now I think it's better minified and served with the rest.

See https://github.com/webcompat/webcompat.com/blob/master/grunt-tasks/concat.js#L8.

File: webcompat/static/css/development/components/qrcode.css (hunk 2 above)
Comment (tagawa) [inline webcompat/static/css/development/components/qrcode.css]: Ah, missed that one. Thanks.

File: webcompat/static/js/lib/qrcode.js (hunk 4 above)
Comment (tagawa) [inline webcompat/static/js/lib/qrcode.js]: Thanks, fixed.

File: webcompat/templates/issue.html (hunk 5 above)
Comment (tagawa) [inline webcompat/templates/issue.html]: Good call. Done.

File: webcompat/static/js/lib/qrcode.js (hunk 3 above)
Comment (tagawa) [inline webcompat/static/js/lib/qrcode.js]: OK, done.

**Comment by tagawa** [comment]
Thanks for the feedback @miketaylr. I added a test for opening and closing the QR code which seems to be OK, but then I can't deliberately make it fail so something's not right. Would appreciate you giving it a try.

**Comment by miketaylr** [comment]
Awesome. On my TODO list for today.

**Comment by miketaylr** [comment]
Hey @tagawa, I pushed your branch to staging but the QR code doesn't show for me. Can you take a look?

e.g., http://staging.webcompat.com/issues/234

I see `ReferenceError: qr is not defined issues.min.js:6:9156` in the devtools console, after pressing q or clicking on the image.

**Comment by tagawa** [comment]
Er, slight, ahem, user error :scream: 
I hadn't noticed the QR code was ignored by `.gitignore`. Now added so hopefully should be OK.

**Comment by miketaylr** [comment]
Ah. Good news is that .gitignore bugs are some of the easier bugs to fix. ^_^

**Comment by miketaylr** [comment]
This looks great @tagawa --a billion apologies for taking so long to land it. I split up the test you wrote into 4 tests, and they're all passing for me locally. Assuming Travis is happy we'll go ahead and merge.

Some things to note: it would be cool if you could close the QR modal with Esc, perhaps. Maybe also worth adding a hint like the one we have for the GitHub warp. But those can be new issues. :+1:

**Comment by miketaylr** [comment]
Travis looks happy https://travis-ci.org/webcompat/webcompat.com/builds/71330928

**Comment by miketaylr** [comment]
(I'll try to get this deployed tomorrow).

**Comment by tagawa** [comment]
Thank you Taylormeister. Re ESC to close the modal, I tried adding this but that key event is bound to the labels modal. There may be a way to only attach the event listener when the modal is displayed but I haven't looked into it.

A subtle warp-like key hint is a good idea.

**Comment by miketaylr** [comment]
>  I tried adding this but that key event is bound to the labels modal.

Ah true. I bet we could write some kind of generic `closeModalThingy` handler that checks if either the label editor or the qr modal is open, and closes whichever one it is. (If there isn't some built in way to scope the shortcuts).

**Comment by miketaylr** [comment]
OK, all deployed now.

**Comment by tagawa** [comment]
Yay, thank you! :cherries:

**Pull request merged**